### DISCUSSION TITLE
Should we have osm2pgsql.relation_member_ids similar to osm2pgsql.way_member_ids?

### DISCUSSION THREAD

**Discussion by caspg** [osm2pgsql-dev/osm2pgsql#2268] [Ideas]
I've created in my lua config `relation_member_ids` based on `osm2pgsql.way_member_ids`. 

Should I open PR with this addition or is there any specific reason why it's not present already?

**Comment by joto** [comment]
Two-stage processing does not work for relations, so I don't see why we would need that?

**Reply by caspg** [reply]
I've used it inside `function osm2pgsql.process_relation(relation)`. Wanted to store member relation ids inside relation for further processing.

If that's not a common use case then maybe it's not needed inside osm2pgsql.

**Reply by joto** [reply]
It would certainly be nice if osm2pgsql could handle relations in relations. But so far nobody I am aware of has looked into what would be needed/sensible for that. What is your use case?

**Reply by caspg** [reply]
I'm working on a cycling app. Some of the routes in OSM are "super routes" (relations in relations with few level of nesting).

* EuroVelo 10 top super relation [OSM](https://www.openstreetmap.org/relation/2689634) [waymarkedtrails](https://cycling.waymarkedtrails.org/#route?id=2689634&type=relation&map=5.0/60.4928/18.8055)
* EuroVelo 10 part Poland super relation [OSM](https://www.openstreetmap.org/relation/9566115)
* EuroVelo 10 part Poland 3 [OSM](https://www.openstreetmap.org/relation/9422510#map=10/54.7333/18.0989)

I want to filter out those "sub routes" and only find super relations. Also I need to build geometry of those super relations.

**Reply by joto** [reply]
There is currently no way to do this with osm2pgsql. So you'll have to import them all into the database and do some post-processing there. We have always been shying away from implementing super relations (or any other thing where relations of relations are involved) due to the complexity of the processing. Especially when updates are involved. Or nesting several levels deep.

**Reply by caspg** [reply]
Yup, this is quite complex task and I couldn't find any tool to handle that. Also it's probably quite niche use case, mostly for cycling and hiking long distance trails.
